subsys/net/lib/downloader/src/transports/http.c

```diff
@@ -145,22 +148,21 @@ static int http_get_request_send(struct downloader *dl)
 	/* nRF91 series has a limitation of decoding ~2k of data at once when using TLS */
 	tls_force_range = (http->sock.proto == IPPROTO_TLS_1_2 && !dl->host_cfg.set_native_tls &&
 			   IS_ENABLED(CONFIG_SOC_SERIES_NRF91X));
-
-	if (dl->host_cfg.range_override) {
-		if (tls_force_range && dl->host_cfg.range_override > (TLS_RANGE_MAX)) {
+	if (tls_force_range) {
+		if (dl->host_cfg.range_override > TLS_RANGE_MAX) {
 			LOG_WRN("Range override > TLS max range, setting to TLS max range");
-			dl->host_cfg.range_override = (TLS_RANGE_MAX);
+			dl->host_cfg.range_override = TLS_RANGE_MAX;
+		} else if (dl->host_cfg.range_override == 0) {
+			dl->host_cfg.range_override = TLS_RANGE_MAX;
```

Comment:
Why no warning here?

Reply:
Because here it is not explicitly set to something else by the application, as it is above.